frontend/src/pages/Register.spec.js

```diff
@@ -220,42 +221,79 @@ describe('Register', () => {
-          expect(wrapper.find('span.alert-text').exists()).toBeTruthy()
-          expect(wrapper.find('span.alert-text').text().length !== 0).toBeTruthy()
-          expect(wrapper.find('span.alert-text').text()).toContain('error.error')
-          expect(wrapper.find('span.alert-text').text()).toContain('Ouch!')
-        })
+        describe('server sends back error "User already exists."', () => {
+          beforeEach(async () => {
+            await createError('GraphQL error: User already exists.')
+          })
+
+          it('shows success title, subtitle, login button', () => {
+            expect(wrapper.vm.showPageMessage).toBe(true)
+            expect(wrapper.find('.test-message-headline').text()).toBe('site.thx.errorTitle')
+            expect(wrapper.find('.test-message-subtitle').text()).toBe('error.user-already-exists')
+            expect(wrapper.find('.test-message-button').text()).toBe(
+              'site.register.message-button-text',
+            )
+          })
+
+          it('toasts the error message', () => {
+            expect(toastErrorSpy).toBeCalledWith('error.user-already-exists')
+          })
 
-        it('button to dismisses error message is present', () => {
-          expect(wrapper.find('button.close').exists()).toBeTruthy()
+          it('click calls "solveError"', async () => {
+            wrapper.find('.test-message-button').trigger('click')
+            await wrapper.vm.$nextTick()
+            expect(wrapper.vm.showPageMessage).toBe(false)
+          })
         })
 
-        it('dismisses error message', async () => {
-          await wrapper.find('button.close').trigger('click')
-          await flushPromises()
-          expect(wrapper.find('span.alert-text').exists()).not.toBeTruthy()
+        describe('server sends back error "Unknown error"', () => {
+          beforeEach(async () => {
+            await createError(' – Unknown error.')
+          })
+
+          it('shows success title, subtitle, login button', () => {
+            expect(wrapper.vm.showPageMessage).toBe(true)
+            expect(wrapper.find('.test-message-headline').text()).toBe('site.thx.errorTitle')
+            expect(wrapper.find('.test-message-subtitle').text()).toBe(
+              'error.unknown-error – Unknown error.',
+            )
+            expect(wrapper.find('.test-message-button').text()).toBe(
+              'site.register.message-button-text',
+            )
+          })
+
+          it('toasts the error message', () => {
+            expect(toastErrorSpy).toBeCalledWith('error.unknown-error – Unknown error.')
+          })
+
+          it('click calls "solveError"', async () => {
+            wrapper.find('.test-message-button').trigger('click')
+            await wrapper.vm.$nextTick()
+            expect(wrapper.vm.showPageMessage).toBe(false)
+          })
         })
       })
 
       describe('server sends back success', () => {
-        beforeEach(() => {
+        beforeEach(async () => {
```

Comment:
Why async?

Reply:
```js
        beforeEach(async () => {
          registerUserMutationMock.mockResolvedValue({
            data: {
              create: 'success',
            },
          })
          await wrapper.find('form').trigger('submit')
          await flushPromises()
        })
```

Because of the `await wrapper.find('form').trigger('submit')` and `await flushPromises()` in it.